runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpStateSingle.java

```diff
@@ -294,6 +296,8 @@ boolean update(long currentXp, int goalStartXp, int goalEndXp)
 			{
 				endLevelExp = goalEndXp;
 			}
+
+			hasGoal = goalEndXp != -1;
```

Comment:
When testing this on my maxed main, I found the value was `0` for skills without a goal set. I'm aware the call originating from `XpTrackerPlugin#onStatChanged()` can supposedly have a `-1` value, so this should be `goalEndXp <= 0` instead.
```suggestion
			hasGoal = goalEndXp <= 0;
```